function-invokers/shell-function-invoker/src/main/docker/image/runner.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+
+set -e
+
+source $FUNCTION_URI
+
+while read X; do echo $(userfunction "$X"); done </pipes/input >/pipes/output
```

Comment:
if we're going to be prescriptive about a function name, maybe it should be "main"?

Reply:
Couldn't the name of the function be an env var?

Reply:
well we could use the same approach as elsewhere (needs to change to either a proper "query param" or a `#` fragment I guess): `FUNCTION_URI=file:///foo/bar.sh?function=whatever`

That would also mean it could be optional, so that the script itself could be invoked directly when the user doesn't want to wrap their logic in a function (as could be the case with the simple `wc` example).